client/my-sites/marketplace/pages/marketplace-plugin-upload-status/index.tsx

```diff
@@ -62,6 +64,31 @@ const MarketplacePluginInstall = ( { productSlug } ): JSX.Element => {
 		getStatusForPlugin( state, siteId, productSlug )
 	);
 
+	const supportsAtomicUpgrade = useRef< boolean >();
+	useEffect( () => {
+		supportsAtomicUpgrade.current =
+			isBusiness( selectedSite?.plan ) ||
+			isEnterprise( selectedSite?.plan ) ||
+			isEcommerce( selectedSite?.plan );
```

Comment:
We would also want to check whether this is 
- a Jetpack site
https://github.com/Automattic/wp-calypso/blob/2f3fc556dfa199386ea5816947910d7ab91904a2/client/my-sites/plugins/plugin-details.jsx#L307
- a VIP site
https://github.com/Automattic/wp-calypso/pull/57902/files

I thinks it's time that we extract this logic to a separate component inside the Marketplace folder which will allow us to know whether a plan supports installing a product (plugins currently, themes in the future) so that we can reuse it in multiple places.

Reply:
The jetpack site is verified later down the line to understand if we want to transfer to atomic or not. 

We still want to verify if this logic works for VIP sites though.

Reply:
I'm having some issues identifying when a site is VIP or not. Those are props I'm getting from the `getSelectedSite` selector on the codep2 site which displays the VIP menu.

```json
"name":"codep2",
"plan":{
  "product_id":1003,
  "product_slug":"value_bundle",
  "product_name_short":"Premium",
  "expired":false,
  "user_is_owner":false,
  "is_free":false
},
"is_vip":false,
...
```

For now I didn't make any changes related to VIP plans.